### PR TITLE
Added parsing of pluggable monitors in package_index.json

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls) before creating one)
- [x] The PR follows [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**
Added functions for parsing pluggable monitor fields in package_index.json.
The added functions are sort of duplicated from the ones that handle pluggable discoveries, I have another PR to factor out the common part but is a bit more involved, so for now I'll push this one so we can move forward.

- **Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**
No

### CODE REVIEW

File: arduino/cores/status.go [hunk 1]
```diff
@@ -127,6 +127,33 @@ func (packages Packages) GetPlatformReleaseDiscoveryDependencies(release *Platfo
...
+
+		// We always want to use the latest available release for monitors
+		latestRelease := tool.LatestRelease()
+		if latestRelease == nil {
+			return nil, fmt.Errorf(tr("can't find latest release of %s"), monitor.Name)
+		}
+		res = append(res, latestRelease)
+	}
+	return res, nil
+}
```
Comment (silvanocerza) [inline arduino/cores/status.go:155]: This is identical to `GetPlatformReleaseDiscoveryDependencies` apart from line 137, we could make generic.

**Review by silvanocerza — CHANGES_REQUESTED** [review]
Small change to make, rest is good.

File: arduino/cores/status.go (hunk 1 above)
Comment (cmaglie) [reply arduino/cores/status.go:155]: Yes, also `MonitorDependencies` is identical to `DiscoveryDependencies`.

The problem is that `release.MonitorDependencies` is an array of `MonitorDependecy`... we can make it generic if we change the type of `release.DiscoveryDependencies` and `release.MonitorDependencies` to be the same type (for example reusing`ToolDependencies` for both) but this change starts a chain reaction... I've another branch ready for that but I prefer to push it in another PR for review and not delay this one.

Comment (silvanocerza) [reply arduino/cores/status.go:155]: Oh right right, the type is different. Let's leave as it is now for the time being.

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**